designdocs/provider_boosting_implementation.md

```diff
@@ -167,7 +167,12 @@ pub trait StakingRewardsProvider<T: Config> {
     /// Returns whether the claim passes validation.  Accounts must first pass `payoutEligible` test.
     /// Errors: None
     fn validate_staking_reward_claim(account_id: T::AccountID, proof: Hash, payload: StakingRewardClaim<T>) -> bool;
-}
+
+    /// Calculate a reward for a single era based on a chosen economic model
+    fn staking_reward_for_era(amount_staked: BalanceOf<T>,
+		total_staked: BalanceOf<T>,
+		reward_pool_size: BalanceOf<T>,
+    ) -> BalanceOf<T>}
```

Comment:
I agree it's not the best name. We're actually not only computing rewards for the current era.  This would be called whenever staking rewards were claimed, so it would be computing rewards for each historical era they've been staking for.  

I'm also out of ideas.

Reply:
Are any of the arguments meant to be era-specific? If so, maybe change their names, like `era_amount_staked`, `era_total_staked`, etc.

Reply:
How about: 

```rust
	/// Calculate the staking reward for a single era.  We don't care about the era number,
	/// just the values.
	fn era_staking_reward(
		era_amount_staked: BalanceOf<T>, // how much individual staked for a specific era
		era_total_staked: BalanceOf<T>,  // how much everyone staked for the era
		era_reward_pool_size: BalanceOf<T>, // how much token in the reward pool that era
	) -> BalanceOf<T>;
```